src/Admin/AbstractAdmin.php

```diff
@@ -1365,6 +1362,12 @@ public function generateMenuUrl($name, array $parameters = [], $referenceType =
      */
     public function setTemplates(array $templates)
```

Comment:
Actually this method is used to define your own templates on a per admin basis (using DI on services.yaml). How do you suggest to do this if you remove it?

Reply:
(Same for setTemplate())

Reply:
> Actually this method is used to define your own templates on a per admin basis (using DI on services.yaml). How do you suggest to do this if you remove it?

```php
public function setTemplate($name, $template)
{
    // NEXT_MAJOR: Remove this line
    $this->templates[$name] = $template;

    $this->getTemplateRegistry()->setTemplate($name, $template);
}
```
In next major (in 4.x):
```php
public function setTemplate($name, $template)
{
    $this->getTemplateRegistry()->setTemplate($name, $template);
}
```

We don't need this `setTemplate` method becouse we can do it directly from template registry. The same for `setTemplates`. Advance for this is that we encapsulete view from admin.

Thanks for ask about it. I should add `next_major` in DI too.

Reply:
BTW I want extract `DemoBundle` from sandbox to standalone bundle. It will be awesome for generate demo fixture for example for Post based on people prefers (sonata_demo config). This solution about how to set own template to admin can be add there.

Reply:
I still do not understand what you mean.

Reply:
Curently for each admin definition DI create own template registry. `some.admin` => `some.admin.template_registry`. People can set own template by call `setTemplate` or `setTemplates`. IMO we should create `sonata.admin.template_registry` and call `setTemplateRegistry` (as default). People should also create own `some.admin.custom.template_registry` and set it as whole object. Based on `sonata.admin.template_registry` tag we can also allow for switchable template registry for each admin. People will be able to (for example) to show standard product view or thumbnail view. This will be not possible when we keep this methods.